test/connect-backbone-to-react.test.js

```diff
@@ -594,6 +594,76 @@ describe('connectBackboneToReact', function() {
     });
   });
 
+  describe('when passed props change to include', function() {
+    let ConnectedTest;
+    let setStateSpy;
+    let createListenerSpy;
+    let removeListenerSpy;
+
+    beforeEach(function() {
+      ConnectedTest = connectBackboneToReact(mapModelsToProps)(TestComponent);
+      wrapper = mount(<ConnectedTest models={modelsMap} />);
+
+      setStateSpy = sandbox.spy(ConnectedTest.prototype, 'setState');
+      createListenerSpy = sandbox.spy(ConnectedTest.prototype, 'createEventListener');
+      removeListenerSpy = sandbox.spy(ConnectedTest.prototype, 'removeEventListener');
+
+      const decoratorUserModel = new UserModel({
+        name: 'Robert',
+        age: '30',
+        hungry: false,
+      });
+
+      const initialModelsMap = {
+        user: userModel,
+        coll: userCollection,
+        decorator: decoratorUserModel,
+      };
+
+      wrapper.setProps({ models: initialModelsMap });
+    });
+
+    afterEach(function() {
+      wrapper.unmount();
+    });
+
+    it('calls setState once', function() {
+      assert.equal(setStateSpy.callCount, 1);
+    });
+
+    it('calls createEventListener once for the model', function() {
+      assert.equal(createListenerSpy.callCount, 1);
+    });
+
+    it('does not call removeEventListener for the model', function() {
+      assert.equal(removeListenerSpy.callCount, 0);
+    });
+
+    describe('an undefined model', function() {
+      beforeEach(function() {
+        const newModelsMap = {
+          user: userModel,
+          coll: userCollection,
+          decorator: undefined,
+        };
+
+        wrapper.setProps({ models: newModelsMap });
+      });
+
+      it('calls setState again', function() {
+        assert.equal(setStateSpy.callCount, 2);
+      });
+
+      it('does not call createEventListener again for the model', function() {
+        assert.equal(createListenerSpy.callCount, 1);
+      });
+
+      it('calls removeEventListener once for the model', function() {
```

Comment:
We should assert that it's being called for the model that we are removing, as that's the behavior we're concerned about.

Reply:
Good idea!